src/UI/Notifications.hs

```diff
@@ -0,0 +1,81 @@
+-- This file is part of purebred
+-- Copyright (C) 2021 Róman Joost
+--
+-- purebred is free software: you can redistribute it and/or modify
+-- it under the terms of the GNU Affero General Public License as published by
+-- the Free Software Foundation, either version 3 of the License, or
+-- (at your option) any later version.
+--
+-- This program is distributed in the hope that it will be useful,
+-- but WITHOUT ANY WARRANTY; without even the implied warranty of
+-- MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+-- GNU Affero General Public License for more details.
+--
+-- You should have received a copy of the GNU Affero General Public License
+-- along with this program.  If not, see <http://www.gnu.org/licenses/>.
+{-# LANGUAGE FlexibleContexts #-}
+module UI.Notifications (
+  -- * Overview
+  -- $overview
+
+  -- * Setters on the AppState
+  setError
+
+  -- * Utilities
+  , hasError
+
+  -- * Factory functions
+  , makeInfo
+  , makeWarning
+  , makeError
+
+  -- * State Monad Setters
+  , showError
+  , showWarning
+  , showInfo
+  , showUserMessage
+  )
+where
+
+import Control.Lens (view, set, assign)
+import qualified Data.Text as T
+import Control.Monad.State
+
+import Types
+import Error
+
+{- $overview
+Notifications are one-line messages shown to the user. That can be
+Errors, Warnings or simply acknowledgement of actions.
+-}
+
+setError :: Error -> AppState -> AppState
+setError = set asUserMessage . Just . makeError StatusBar
+
+hasError :: AppState -> Bool
+hasError = go . view asUserMessage
+  where
+    go (Just (UserMessage _ (Error _))) = True
+    go _ = False
+
+showError :: (MonadState AppState m) => Error -> m ()
+showError = showUserMessage . makeError StatusBar
+
+showWarning :: (MonadState AppState m) => T.Text -> m ()
+showWarning = showUserMessage . makeWarning StatusBar
+
+showInfo :: (MonadState AppState m) => T.Text -> m ()
+showInfo = showUserMessage . makeInfo StatusBar
+
+showUserMessage :: (MonadState AppState m) => UserMessage -> m ()
+showUserMessage = assign asUserMessage . Just
+
+
+makeInfo :: Name -> T.Text -> UserMessage
+makeInfo n = UserMessage n . Info
+
+makeWarning :: Name -> T.Text -> UserMessage
+makeWarning n = UserMessage n . Warning
+
+makeError :: Name -> Error -> UserMessage
+makeError n = UserMessage n . Error
```

Comment:
The functions dealing with `Error` are a bit "odd one out", with the others dealing with `Text`.  Not a criticism really, just an observation.   It might be nice to have a shortcut function such as:
```haskell
makeGenericError :: Name -> T.Text -> UserMessage
makeGenericError n = UserMessage n . Error . GenericError
```

Reply:
Probably time to change the argument of `GenericError` constructor from `String` to `Text` too.

Reply:
Hm.. see I think we should throw out the `GenericError`.. I've hopefully changed all the `GenericError`s to warnings and IMHO either we have a specific error to show or we use a different severity. But I'd say there isn't really something like a "generic" error...

Reply:
Yeah I guess the `Error` still sticks out as odd, but I hope I made it less weird. Also removed the `GenericError`. Either we use a special `Error` constructor or it's a different severity.